### PR TITLE
Bugfix resource caching

### LINKED ISSUE #300: Cached resource version is never used
There are two possible cases for caching:
1) use the `Downloader` without a specified cache dir (this case is tested in the corresponding unit test)
2) use the `Downloader` with a specified cache dir (this case is not tested in the corresponding unit test)

When running the unit test with a `Downloader` with a specified cache dir, they fail.

Reasons: 
- The corresponding else branch, if the cached resource should be used, is missing [https://github.com/nilskre/biocypher/blob/main/biocypher/_get.py#L124](https://github.com/nilskre/biocypher/blob/main/biocypher/_get.py#L124). Therefore the cached resource is never used.
- In the test case 1) every time a new tmp cache dir is created. Thus the reading process of a cached resource is never executed, nor tested.

### PR DESCRIPTION
**Pull request by nilskre**
Fix #300

### CODE REVIEW

**Comment by slobentanzer** [comment]
Thanks a lot; there are some issues that prevent a merge, and due to the hackathon, I did a manual fix of the initial problem on `main` (the conversion between datetime and str was implemented inconsistently). We can get back to the PR after the hackathon.

File: test/test_get.py [hunk 1]
```diff
@@ -10,8 +10,18 @@
...
+
+
+@pytest.fixture
+def downloader_without_specified_cache_dir():
+    return Downloader()
+
+
+@pytest.fixture
+def downloader_with_specified_cache_dir():
+    return Downloader(cache_dir="./.cache")
```
Comment (slobentanzer) [inline test/test_get.py:24]: tests should not write to the project directory. for keeping a folder around for multiple tests, we usually use pytest fixture scopes (module or session scope). can get tricky though. :)

File: test/test_get.py [hunk 2]
```diff
@@ -46,20 +72,35 @@ def test_download_file(downloader):
...
     paths = downloader.download(resource)
-    # should not download again
-    assert paths[0] is None
-
-    # manipulate cache dict to test expiration (datetime format)
-    downloader.cache_dict["test_resource"][
-        "date_downloaded"
-    ] = datetime.now() - timedelta(days=8)
+    # should not download again (assert that cache folder is returned)
+    if downloader == "downloader_without_specified_cache_dir":
```
Comment (slobentanzer) [inline test/test_get.py:76]: these conditionals both fail, because downloader is an instance of Downloader, not a string

File: test/test_get.py [hunk 3]
```diff
@@ -46,20 +72,35 @@ def test_download_file(downloader):
...
+        datetime.now() - timedelta(days=8)
+    )
 
     paths = downloader.download(resource)
     # should download again
     assert len(paths) == 1
-    assert paths[0] is not None
-
-
+    if downloader == "downloader_without_specified_cache_dir":
```
Comment (slobentanzer) [inline test/test_get.py:89]: similar to above, both conditionals fail here, never asserting anything

File: test/test_get.py (hunk 2 above)
Comment (nilskre) [reply test/test_get.py:76]: True. :see_no_evil: Learning: never trust what ChatGPT is generating

**Comment by nilskre** [comment]
Updated this PR based on the feedback and now ready for second review.

**Comment by nilskre** [comment]
The [test_download_directory_and_caching](https://github.com/biocypher/biocypher/blob/main/test/test_get.py#L127) and [test_download_zip_and_expiration](https://github.com/biocypher/biocypher/blob/main/test/test_get.py#L150) are not using the `Downloader` fixture. Is this intended?

**Pull request merged**